src/Compiler/Microsoft.AspNetCore.Razor.Language/src/CodeGeneration/CodeWriterExtensions.cs

```diff
@@ -538,7 +539,18 @@ public static IDisposable BuildLinePragma(this CodeWriter writer, SourceSpan? sp
             return NullDisposable.Default;
         }
 
-        return new LinePragmaWriter(writer, span.Value, context, 0, false);
+        var spanValue = span.Value;
+        if (context.Options.RemapDesignTimeLinePragmaPathsOnWindows && RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+        {
+            // ISSUE: https://github.com/dotnet/razor/issues/9108
+            // The razor tooling normalizes paths to be forward slash based, regardless of OS.
+            // If you try and use the line pragma in the design time docs to map back to the original file it will fail,
+            // as the path isn't actually valid on windows. As a workaround we apply a simple heuristic to switch the
+            // paths back when writing out the design time paths.
+            spanValue = new SourceSpan(spanValue.FilePath.Replace("/", "\\"), spanValue.AbsoluteIndex, spanValue.LineIndex, spanValue.CharacterIndex, spanValue.Length);
```

Comment:
Can there be a path with leading slash like `\\SERVER\user\file.razor`? Should we also do `.TrimStart('\\')` like tooling does?

https://github.com/dotnet/razor/blob/c3b3fab48624fc4559d2c38535d7b7c3d0cfe176/src/Razor/src/Microsoft.AspNetCore.Razor.ExternalAccess.RoslynWorkspace/RazorProjectInfoSerializer.cs#L187-L188

Reply:
If you do that, please switch over to a span to avoid the extra string allocation.

Reply:
I don't think we need to do this. I've added a test case covering the server path versions too.